src/dlload.c

```diff
@@ -139,26 +136,20 @@ static void *jl_load_dynamic_library_(const char *modname, unsigned flags, int t
 #endif
         goto done;
     }
-    /*
-      this branch shortcuts absolute paths
-    */
+
 #ifdef _OS_WINDOWS_
-    else if (modname[1] == ':') {
+    abspath = modname[0] == '/' || modname[0] == '\\' ||
+        (modname[0] && modname[1] == ':' && (modname[2] == '\\' || modname[2] == '/'));
 #else
-    else if (modname[0] == '/') {
+    abspath = modname[0] == '/';
```

Comment:
init.c has an `isabspath` function. We might want to move that somewhere where it can be used more widely.